sound/soc/sof/sof-of-dev.c

```diff
@@ -45,7 +45,18 @@ static struct sof_dev_desc sof_of_imx8mp_desc = {
 };
 #endif
 
+static int snd_of_prepare(struct device *dev)
+{
+	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
+
+	sdev->system_suspend_target = SOF_SUSPEND_S3;
+
+	return 0;
+}
+
```

Comment:
is this needed? It's the same as what snd_sof_prepare() does - or do you have CONFIG_ACPI enabled on i.MX?

Reply:
For some reason we have CONFIG_ACPI enabled. I think we inherit it from generic arm64 defconfig.

Reply:
make we should have a helper that sets the value to S3 only, and a more fancy one that deals with the acpi_target_state().
@ranj063 what do you think?

Reply:
@plbossart thats what I was thinking too but how do we decide which one to use? Should we base it on the platform config?

Reply:
Not sure what you meant by platform config @ranj063 ? the point is that CONFIG_ACPI is not good enough, it's used in ARM platforms so can't be used to select Intel-specific code.

Reply:
I meant the configs for each platform for ex: CONFIG_SND_SOC_SOF_BAYTRAIL and the others but that would be too much I think. Can we use these?
CONFIG_SND_SOC_SOF_PCI
CONFIG_SND_SOC_SOF_ACPI
CONFIG_SND_SOC_SOF_OF

Reply:
Maybe improve granularity, in case of `snd_sof_prepare()` and S0IX support in general, add a check - run-time or compile-time - for whether it's supported?

Reply:
@lyakh, initially I thought on adding a new field ```use_acpi``` in ```snd_sof_dev```. But I think your suggestion is better. 
Add a field ```pm_caps``` inside ```snd_sof_dev``` for supported PM states.

Reply:
I don't think we want to hard-code power states, they might be dependent on firmware (BIOS) enabling specific states for e.g. PCI.